### DISCUSSION TITLE
How to change OAuth rules.

### DISCUSSION THREAD

**Discussion by AaronHessler** [pocketbase/pocketbase#2324] [Q&A] [answered]
I have noticed that my new users are automatically verified. Is there any way of changing that?
There don't seem to be settings for that.

I use Microsoft OAuth.

**Comment by ganigeorgiev** [comment, marked as answer]
The default behavior could be changed if you attach to the [`OnRecordBeforeAuthWithOAuth2Request` hook](https://pocketbase.io/docs/event-hooks/#record-auth-api-hooks).

OAuth2 users are marked as `verified` because it is expected the OAuth2 provider to be a trusted source.

**Comment by Plorenzo** [comment]
>OAuth2 users are marked as verified because it is expected the OAuth2 provider to be a trusted source.

@ganigeorgiev is this still the case in v0.23? 

My google oauth accounts are being created with email `verified=fasle` (I may be doing something wrong again)

how would you check if a new user record is being created from a oauth request on a hook? to then manually set `verified` to true
